pkg/cri/server/server.go

```diff
@@ -342,6 +342,7 @@ func (s *server) intercept(ctx context.Context, req interface{},
 	}
 
 	s.collectStatistics(kind, name, start, send, recv, end)
+	logger.Flush()
```

Comment:
Hmm, should be `defer logger.Flush()` instead? Would flush in case of errors etc.

Reply:
That's what I head initially. But because the whole function is obviously deliberately written so that it never return anywhere else than in the end, I changed it to the current form.

Reply:
But maybe it would be less clobbersome if we move the log flush to the end of server.interceptRequest() (the caller of inercept), or to the beginning of that with a deferred call. WDYT?

Reply:
> the whole function is obviously deliberately written so that it never returns anywhere else than in the end

Fair enough, makes sense. Just a smartass note that this might be changed by a future patch 😉 



> But maybe it would be less clobbersome if we move the log flush to the end of server.interceptRequest() (the caller of inercept), or to the beginning of that with a deferred call.

Dunno, I'm really undecided on that. I think the current location is good but don't object if you want to move it